test/index.test.ts

```diff
@@ -293,7 +405,191 @@ describe('HipThrusTS', () => {
+        const aPassedIn = 'some string';
+        const bReturned = 4;
+        const cReturned = 6;
 
+        async function HTPipeTypeMismatchTest() {
+          await HTPipeTest(
+            {
+              attachData: (context: { a: string }) => {
+                return {
+                  c: cReturned,
+                  b: bReturned,
+                };
+              },
+            },
+            'attachData',
+            { a: aPassedIn },
+            { b: bReturned },
+            // @ts-expect-error
+            true
+          );
+        }
+      });
+      it('should give error if pipeIn has same count of keys as lifecycle stage context in, and pipe out has same count of keys as lifecycle stage context out, but pipeIn key values has different types', () => {
+        const incorrectAValue = 3;
+        const bReturned = 4;
+
+        async function HTPipeTypeMismatchTest() {
+          await HTPipeTest(
+            {
+              attachData: (context: { a: string }) => {
+                return {
+                  b: bReturned,
+                };
+              },
+            },
+            'attachData',
+            { a: incorrectAValue },
+            { b: bReturned },
+            // @ts-expect-error
+            true
+          );
+        }
+      });
+      it('should give error if pipeIn has same count of keys as lifecycle stage context in, and pipe out has same count of keys as lifecycle stage context out, but pipeOut key values has different types', () => {
+        const aPassedIn = 'some string';
+        const bReturned = 4;
+
+        async function HTPipeTypeMismatchTest() {
+          await HTPipeTest(
+            {
+              attachData: (context: { a: string }) => {
+                return {
+                  b: bReturned,
+                };
+              },
+            },
+            'attachData',
+            { a: aPassedIn },
+            { b: 'string' },
+            // @ts-expect-error
+            true
+          );
+        }
+      });
+      it('should give error if pipeIn and PipeOut have correct types, but return data of lifecycle stage have different values than pipeOut', async () => {
+        const aPassedIn = 'some string';
+        const bReturned = 4;
+        const cReturned = 6;
+        const incorrectCReturned = 12;
+
+        let errorReturned;
+
+        try {
+          await HTPipeTest(
+            {
+              attachData: (context: { a: string }) => {
+                return {
+                  c: cReturned,
+                  b: bReturned,
+                };
+              },
+            },
+            'attachData',
+            { a: aPassedIn },
+            { b: bReturned, c: incorrectCReturned },
+            true
+          );
+        } catch (err) {
+          errorReturned = err;
+        }
+        // tslint:disable-next-line:no-unused-expression
```

Comment:
are you sure this try/catch construct does what you expect??  For example, if you comment out all the expect() calls in HTPipeTest(), will this test fail?  I have a hunch that it won't.
- [ ] confirm this does what you think.

Reply:
yeah it's do what i want but it give wrong error message
so i decide to change it a little
expect(errorReturned).to.exist; will give error if errorReturned is undefined or null